src/dataflow/src/sink/tail.rs

```diff
@@ -159,51 +138,36 @@ fn tail<G>(
                 }
             }
 
-            // Sort results by time and convert to Vec<Row>. We use stable sort here because
-            // it will produce deterministic results since the cursor will always produce
-            // rows in the same order.
-            results.sort_by_key(|(time, _)| *time);
-            let mut results: Vec<Row> = results.into_iter().map(|(_, row)| row).collect();
-
-            if let Some(batch) = batches.last() {
-                let progress_row = update_progress(
-                    &mut input_frontier,
-                    batch.desc.upper().borrow(),
-                    &mut packer,
-                    connector.object_columns + 1,
-                );
-                if connector.emit_progress && tail_protocol.is_some() {
-                    if let Some(progress_row) = progress_row {
-                        results.push(progress_row);
-                    }
-                }
-            }
+            // Sort results by time. We use stable sort here because it will produce deterministic
+            // results since the cursor will always producerows in the same order.
+            results.sort_by_key(|(time, _, _)| *time);
```

Comment:
Is this important to do? If it is, we might want to commit to it in the description of the response variant. If it isn't important (or we don't want to commit to it) the perhaps leave a note / move the code to the coordinator? I'm not sure which one is the answer, though.